WooCommerce/WooCommerceTests/ViewRelated/HubMenu/HubMenuViewModelTests.swift

```diff
@@ -518,6 +518,73 @@ final class HubMenuViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.navigationPath, NavigationPath([HubMenuNavigationDestination.payments]))
     }
 
+    @MainActor
+    func test_navigateToDestination_replaces_navigationPath_with_specified_destination() {
+        // Given
+        let navigationPath = NavigationPath(["testPath1", "testPath2"])
+        let viewModel = HubMenuViewModel(siteID: sampleSiteID,
+                                         tapToPayBadgePromotionChecker: TapToPayBadgePromotionChecker())
+        viewModel.navigationPath = navigationPath
+        XCTAssertEqual(viewModel.navigationPath.count, 2)
+
+        let expectedMenusAndDestinations: [HubMenuNavigationDestination: HubMenuItem] = [
+            .settings: HubMenuViewModel.Settings(),
+            .payments: HubMenuViewModel.Payments(),
+            .inAppPurchase: HubMenuViewModel.InAppPurchases(),
+            .subscriptions: HubMenuViewModel.Subscriptions(),
+            .blaze: HubMenuViewModel.Blaze(),
+            .wooCommerceAdmin: HubMenuViewModel.WoocommerceAdmin(),
+            .viewStore: HubMenuViewModel.ViewStore(),
+            .coupons: HubMenuViewModel.Coupons(),
+            .reviews: HubMenuViewModel.Reviews(),
+            .inbox: HubMenuViewModel.Inbox(),
+            .customers: HubMenuViewModel.Customers(),
+            .pointOfSales: HubMenuViewModel.PointOfSaleEntryPoint()
+        ]
```

Comment:
Is there a way to make the destinations CaseIterable for this, so that we don't have to remember to add new ones? If not, perhaps just an assert that the count of the cases is what we expect, so that we get a reminder to add them when it changes.

Reply:
Since one of the cases has an associated value, the enum cannot be auto-synthesized when conforming to `CaseIterable`. Implementing a custom method for this protocol conformance goes against what we want, so I followed your other suggestion to add an assert [c02698575895334b3f44e85045d2d125ae79f23c] for the number of cases instead.